### ISSUE TITLE
404 on other languages then the default

### ISSUE THREAD

**Issue by felixmosh** [isaachinman/next-i18next#497]
## Describe the bug
This is my config:
```
{
  browserLanguageDetection: false,
  serverLanguageDetection: false,
  defaultLanguage: 'he',
  otherLanguages: ['en'],
  localeSubpaths: { en: 'en' },
}
```
When I navigate to `/hello-world` -> it loads he, as it is my default lang.
Then when I navigate to `/en/hello-world` -> return 404.

I don't think that this bug is related to #469 because it is the opposite scenario.

## Occurs in next-i18next version
1.1.2

## Expected behaviour
It changes the language to `en`.

## OS (please complete the following information)
 - Device: MBP 15
 - Browser: all

## Additional context
From my debugging looks like this `subpathIsRequired` check in [here](https://github.com/isaachinman/next-i18next/blob/master/src/middlewares/next-i18next-middleware.ts#L49) it not correct cause `req.i18n.language = 'he'` and it is not exists in the `localeSubpaths` map.

I've tried to add `he: ''` to the `localeSubpaths` map, but then this [regex](https://github.com/isaachinman/next-i18next/blob/master/src/middlewares/next-i18next-middleware.ts#L22) gets build with empty string | lang, and the empty string catches all.

**Comment by isaachinman** [comment]
Looks like a different manifestation of the same root cause. Contributions welcome.

**Comment by isaachinman** [comment]
Hi there - could those users experiencing this problem try testing their app against #498?

**Comment by felixmosh** [comment]
@isaachinman looks like it solved the issue at my side 🙏🏼

**Comment by lazidoca** [comment]
I just want to confirm that the fix #498 has solved my similar problem. I got 404 when manually enter URL with locale subpath.
